src/components/VisualSkeletalAnimation.cpp

```diff
@@ -298,7 +300,13 @@ namespace REGoth
   {
     bs::HAnimationClip clip;
 
-    if (Animation::isTransitionNeeded(state))
+    throwIfNotReadyForRendering();
+
+    if (!mSubAnimation->isPlaying())
+    {
+      clip = findAnimationClip(state);
```

Comment:
Haven't digged into how animations work but isn't that `if`-case already handled with the `else`? It's the same code

Reply:
Not quite. If there is no animation playing, then `getStateFromPlayingAnimation()` will return an empty string on the next branch, thus, no transition animation can be found. I didn't want to add conditions to the next `if` to make it clear what is supposed to happen. I'll refactor it!